eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/boot/EventMeshServer.java

```diff
@@ -101,18 +122,26 @@ public void start() throws Exception {
         if (eventMeshHttpConfiguration != null && eventMeshHttpConfiguration.eventMeshServerSecurityEnable) {
             acl.start();
         }
-
+        // registry start
         if (eventMeshTCPConfiguration != null
-                && eventMeshTCPConfiguration.eventMeshTcpServerEnabled
-                && eventMeshTCPConfiguration.eventMeshServerRegistryEnable) {
+            && eventMeshTCPConfiguration.eventMeshTcpServerEnabled
+            && eventMeshTCPConfiguration.eventMeshServerRegistryEnable) {
+            registry.start();
+        }
+        if (eventMeshHttpConfiguration != null && eventMeshHttpConfiguration.eventMeshServerRegistryEnable) {
+            registry.start();
+        }
+        if (eventMeshGrpcConfiguration != null && eventMeshGrpcConfiguration.eventMeshServerRegistryEnable) {
             registry.start();
         }
```

Comment:
Can we simplify this? When we need to register, we may need to register not matter it use http/tcp/grpc.

Reply:
Here we just reuse the existing format and register for each protocol

Reply:
You can split it if you want to start HTTP or GRPC later